src/main.c

```diff
@@ -460,13 +451,25 @@ selected_choice(void)
 static void
 filter_choices(void)
 {
-	struct choice	*choice;
+	int	i;
 
-	SLIST_FOREACH(choice, choices, choices) {
-		choice->score = score(choice->string);
-	}
+	for (i = 0; i < (ssize_t)choices.nmemb; ++i)
+		choices.v[i].score = score(choices.v[i].string);
+
+	qsort(choices.v, choices.nmemb, sizeof(struct choice), sort_choices);
+}
+
+static int
+sort_choices(const void *p1, const void *p2)
```

Comment:
What do you think about naming this something like `compare_choices` or `cmp_choices` to denote that it serves as a comparison function?